x/ecocredit/basket/msg_create.go

```diff
@@ -56,9 +56,13 @@ func (m MsgCreate) ValidateBasic() error {
 	return m.Fee.Validate()
 }
 
-// Validate additional validation with access to the state data.
-func (m MsgCreate) Validate(minFee sdk.Coin) error {
-	// TODO: update function to do stateful validation
+// ValidateCreateFee additional validation with access to the state data.
+// minFee must be sorted.
+func ValidateCreateFee(m *MsgCreate, minFee sdk.Coins) error {
```

Comment:
so the idea was to make all validation here. I break down the some parts of the Create RPC in the keeper methods. Eg `indexAllowedClasses`.

Reply:
in general I would prefer that this code just lives in the `Create` method itself or at least in the same package rather than calling into a separate package. remember that in the future we won't be able to define any code in the api module where the types are generated